core/controllers/base.py

```diff
@@ -187,16 +184,14 @@ def __init__(self, request, response):  # pylint: disable=super-init-not-called
         self.role = (
             feconf.ROLE_ID_GUEST
             if self.user_id is None else user_settings.role)
-        self.actions = role_services.get_all_actions(self.role)
+        self.user = user_services.UserActionsInfo(self.user_id)
 
         self.is_super_admin = (
             current_user_services.is_current_user_super_admin())
 
-        self.values['is_moderator'] = bool(
-            self.role == feconf.ROLE_ID_MODERATOR or
-            self.role == feconf.ROLE_ID_ADMIN)
-        self.values['is_admin'] = bool(
-            self.role == feconf.ROLE_ID_ADMIN)
+        self.values['is_moderator'] = user_services.is_at_least_moderator(
```

Comment:
Then this would be passed all around with the user object as well. Should I do this?
It is used for this purpose (checking if user is at least moderator) here only.